var/spack/repos/builtin/packages/postgresql/package.py

```diff
@@ -103,6 +105,9 @@ def configure_args(self):
         if spec.satisfies("+xml"):
             args.append("--with-libxml")
 
+        if not spec.satisfies("+icu"):
+            args.append("--without-icu")
+
```

Comment:
```suggestion
        if spec.satisfies("~icu"):
            args.append("--without-icu")

```

Quick modification. I think we'll actually want to check `~icu` instead of `not +icu` since the latter will be true for versions < 16 before the variant exists.